dolfyn/rotate/base.py

```diff
@@ -54,164 +54,165 @@ def _check_rotmat_det(rotmat, thresh=1e-3):
-            advo.rotate2('earth', inplace=True)
-            rotation_done_flag = True
-
-    if 'heading' in odata and \
-       not advo.props.get('declination_in_heading', False):
-
-        if cs == 'earth' and not rotation_done_flag:
-            # Rotate to instrument coordinate-system before adjusting
-            # for declination.
-            advo.rotate2('inst', inplace=True)
-
-        odata['heading'] += advo.props['declination']
-        odata['heading'] %= 360
-        advo.props['declination_in_heading'] = True
-
-        if cs == 'earth' and not rotation_done_flag:
-            # Now rotate back to the earth coordinate-system with the
-            # declination included.
-            advo.rotate2('earth', inplace=True)
-    advo.props.pop('__checking_declination__')
-
-
-def nortek_euler2orient(pitch, roll, heading, units='degrees'):
-    # THIS IS FOR NORTEK data ONLY!
-
-    # Heading input is clockwise from North
-    # Returns a rotation matrix that rotates earth (ENU) -> inst.
-    # This is based on the Nortek `Transforms.m` file, available in
-    # the refs folder.
+def euler2orient(heading, pitch, roll, units='degrees'):
+    """
+    Calculate the orientation matrix from euler angles.
+
+    Parameters
+    ----------
+    heading : np.ndarray (Nt)
+      The heading angle of the ADV (clockwise from North).
+    pitch : np.ndarray (Nt)
+      The pitch angle of the ADV.
+    roll : np.ndarray (Nt)
+      The pitch angle of the ADV.
+    units : string {'degrees' (default), 'radians'}
+
+    Returns
+    =======
+    omat : np.ndarray (3x3xNt)
+      The orientation matrix of the data. The returned orientation
+      matrix obeys the following conventions:
+
+       - a "ZYX" rotation order. That is, these variables are computed
+         assuming that rotation from the earth -> instrument frame happens
+         by rotating around the z-axis first (heading), then rotating
+         around the y-axis (pitch), then rotating around the x-axis (roll).
+
+       - heading is defined as the direction the x-axis points, positive
+         clockwise from North (this is *opposite* the right-hand-rule
+         around the Z-axis)
+
+       - pitch is positive when the x-axis pitches up (this is *opposite* the
+         right-hand-rule around the Y-axis)
+
+       - roll is positive according to the right-hand-rule around the
+         instument's x-axis
+
+    """
     if units.lower() == 'degrees':
         pitch = np.deg2rad(pitch)
         roll = np.deg2rad(roll)
         heading = np.deg2rad(heading)
-    # I've fixed the definition of heading to be consistent with
-    # typical definitions.
-    # This also involved swapping the sign on sh in the def of omat
-    # below from the values provided in the Nortek Matlab script
-    heading = (np.pi / 2 - heading)
+    elif units.lower() == 'radians':
+        pass
+    else:
+        raise Exception("Invalid units")
+
+    heading = np.pi / 2 - heading
```

Comment:
This converts the dolfyn-defined heading, where heading is degrees clockwise from +x axis, to the
definition of heading needed for the creation of the orientation matrix, where heading is degrees counterclockwise from the +y axis.

Reply:
Will edit comments on this function in my heading-bug-fix-mcf2 branch. Expect pull request in future.